rfcs/0003-data-storage.md

```diff
@@ -0,0 +1,119 @@
+## Data Storage Structure
+
+![](assets/data-storage.drawio.svg)
+
+### Core Data Types
```

Comment:
These don't seem "types" but "states". Perhaps we can sort them out as the Raft paper into:

* Persistent state on all servers:
* Volatile state on all servers:
* Volatile state on leaders:

And people familiar with Raft will find it easy to follow.

Reply:
Or you're telling the type alias here, and we can have another section to sort out the state.

Reply:
Only the `Term` here represents server state in Raft. Rather than `term` being stored alone, it's actually part of a `Vote{ term, voted_for}` struct that will be covered in a future chapter.

Server states matching the categories you listed will be defined in a separate chapter.